ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -145,11 +162,21 @@ private void reparseAndSuperAnalyze(ASTNode tree, Table mTable, ASTNode tabNameN
       where = (ASTNode)children.get(whereIndex);
       assert where.getToken().getType() == HiveParser.TOK_WHERE :
           "Expected where clause, but found " + where.getName();
-    }
 
-    // Add a sort by clause so that the row ids come out in the correct order
-    appendSortBy(rewrittenQueryStr, columnAppender.getSortKeys());
+      if (shouldOverwrite) {
+        assert where.getChildCount() < 2 : "Overwrite mode not supported with more than 1 children in where clause.";
```

Comment:
assert keyword won't be executed unless you provide "-ea" to the argument list when initializing your JVM

Reply:
Looks like the `assert` keyword is heavily used in SemanticAnalyzer. 
Those statements have a well-defined syntax by the standard and the AST parser builds the tree in a deterministic way. 
Maybe it wasn't worth testing this in prod since this is determined at java compile time and is already covered with end-to-end tests (q tests).
Leaving it up to you ...